jsettlers.logic/src/main/java/jsettlers/logic/movable/MovableWrapper.java

```diff
@@ -0,0 +1,256 @@
+package jsettlers.logic.movable;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+
+import jsettlers.algorithms.path.Path;
+import jsettlers.common.material.EMaterialType;
+import jsettlers.common.movable.EDirection;
+import jsettlers.common.movable.EMovableAction;
+import jsettlers.common.movable.EMovableType;
+import jsettlers.common.position.ShortPoint2D;
+import jsettlers.common.selectable.ESelectionType;
+import jsettlers.logic.buildings.military.IBuildingOccupyableMovable;
+import jsettlers.logic.buildings.military.occupying.IOccupyableBuilding;
+import jsettlers.logic.movable.components.AnimationComponent;
+import jsettlers.logic.movable.components.AttackableComponent;
+import jsettlers.logic.movable.components.GameFieldComponent;
+import jsettlers.logic.movable.components.MaterialComponent;
+import jsettlers.logic.movable.components.MovableComponent;
+import jsettlers.logic.movable.components.PlayerCmdComponent;
+import jsettlers.logic.movable.components.SelectableComponent;
+import jsettlers.logic.movable.components.SpecialistComponent;
+import jsettlers.logic.movable.components.SteeringComponent;
+import jsettlers.logic.movable.interfaces.IAttackable;
+import jsettlers.logic.movable.interfaces.ILogicMovable;
+import jsettlers.logic.player.Player;
+
+/**
+ * @author homoroselaps
+ */
+
+public final class MovableWrapper implements ILogicMovable, Serializable {
+    private static final long serialVersionUID = -2853861825853788354L;
+    private final Entity entity;
+
+    public MovableWrapper(Entity entity) {
+        this.entity = entity;
+    }
+
+    private void readObject(ObjectInputStream ois) throws IOException, ClassNotFoundException {
+        ois.defaultReadObject();
+        entity.get(GameFieldComponent.class).getMovableMap().put(entity.getID(), this);
+        entity.get(GameFieldComponent.class).getAllMovables().add(this);
+    }
+
+    //region Interface Implementations
+    @Override
+    public short getViewDistance() {
+        if (!entity.containsComponent(MovableComponent.class)) return 0;
+        return entity.get(MovableComponent.class).getViewDistance();
```

Comment:
I would change these methods to directly do the get on entity and then check if the result is null. If so, return the default value, if it isn't null, call the method. This way, the code does not need to query the set twice.